src/mlpack/methods/ann/layer/parametric_relu_impl.hpp

```diff
@@ -0,0 +1,140 @@
+/**
+ * @file methods/ann/layer/parametric_relu_impl.hpp
+ * @author Prasanna Patil
+ *
+ * Definition of PReLU layer first introduced in the,
+ * Kaiming He, Xiangyu Zhang, Shaoqing, Ren Jian Sun,
+ * "Delving Deep into Rectifiers:
+ * Surpassing Human-Level Performance on ImageNet Classification", 2014
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_PRELU_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_PRELU_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "parametric_relu.hpp"
+
+namespace mlpack {
+
+template<typename MatType>
+PReLUType<MatType>::PReLUType(const double userAlpha) :
+    Layer<MatType>(),
+    userAlpha(userAlpha)
+{
+  alpha.set_size(WeightSize(), 1);
+  alpha(0) = userAlpha;
```

Comment:
In the new API this is actually not valid---when the layer object is constructed, its trainable weights are not valid.  In fact even the size given by `WeightSize()` is not valid.  Everything about the layer is invalid until `ComputeOutputDimensions()` and then `SetWeightsPtr()` is called.  So, the process of setting the weight value in `alpha` can't be done until `SetWeightsPtr()`.